common/tls.go

```diff
@@ -85,3 +89,45 @@ func RootCAs() *x509.CertPool {
 	}
 	return p
 }
+
+type TrustBundleResponse struct {
+	Certificate string `json:"certificate"`
+}
+
+// TrustBundle root CA certificates pool for connecting to EdgeHub Gateway.
+//
+// The TrustBundle is generated by the Edge Runtime at startup. The connection
+// to the edgeHub (proxy for IoT Hub) requires this cert. This CA cert can be
+// pulled from an API exposed on the edge runtime detailed in the below function.
+func TrustBundle(workloadUri, apiVersion string) *x509.CertPool {
+
+	// format uri string
+	uri := fmt.Sprintf("%strust-bundle?api-version=%s", workloadUri, apiVersion)
+
+	// get http response and handle error
+	resp, err := http.Get(uri)
+	if err != nil {
+		panic("tls: unable to append certificates")
```

Comment:
Could you return a custom error instead of panicking here and bellow in this function, this is not something to panic over :smile: